ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,256 @@
+```
+{
+    "result": "ERROR",
+    "timestamp": "1649842004"
+}
+```
+
+##### Proposed Adapted Tekton Result Format for Xunit Tests Results
+
+- Existing Mapped:
+
+  - **Namespace** == **name** attribute of all of the **TestSuite** elements 
+
+  - **Failures** == **name** attribute for all the **TestCase** elements that have the attribute **status=Failed**
+
+- New Key:
+
+  - **Skipped** == the number of skipped test in integer format from the **TestSuite** attribute **skipped** or from **TestCase** attribute **skipped**
+
+Some Junit XML files generate a **TestSuites** xml element that contains N number of **TestSuite** elements. In this case it maybe simply best to augment our Task Result Schema to be a JSON List of Task Results 
+
+Example output
+```
+[
+    {
+        "result": "FAILURE",
+        "namespace": "Red Hat App Studio E2E tests",
+        "timestamp": "1649148140",
+        "successes": 75,
+        "skipped": 14,
+        "note": "",
+        "failures": [
+            "[It] [release-suite test-demo] Creation of the 'Happy path' resources Create an ApplicationSnapshot."
+        ]
+    },
+    {
+        "result": "SUCCESS",
+        "timestamp": "1649843611",
+        "namespace": "Demo Tests",
+        "successes": 12,
+        "skipped": 0,
+        "note": "",
+        "failures": []
+    }
+]
+```
+
+#### Full Test Output JSON
+
+The Test Tekton tasks will use a standardized manner of displaying testing information. This information will be saved in the form of a JSON file. The contents of this file will relay results of validation by the [Open Policy Agent](https://www.openpolicyagent.org/) policies as executed by the [Conftest](https://www.conftest.dev/) tool and saved in its JSON output format.
+
+Each test will have a standardized short name in snake case, e.g. `release_label_required`, `architecture_label_required`, `architecture_label_deprecated` etc.
+
+The output will provide the following information about the overall test result:
+- **filename** - Name of the file that was inspected by the Tekton task
+  - This will reflect the data that the test was executed on, such as `image-inspect.json`, `clair-vulnerabilities.json` etc.
+- **namespace** - The rego namespace of the test policy. 
+- **successes** - The number of successful checks in the form of an integer
+- **failures** - A JSON list containing objects describing each failure
+
+The **failures** list of objects will provide the following information about each check:
+- **name** - Name of the individual check written in [snake case](https://en.wikipedia.org/wiki/Snake_case)
+- **msg** - Message about what went wrong with the check
+- **description** - The explanation about why the check is significant (Why the label is used, why the vulnerabilities are checked etc.)
+- **url** - link to the further documentation on the individual check (or that type of check)
+
+Example full JSON test output for a Tekton task that tests the container image labels:
```

Comment:
Do you not expect skips, error, and warnings to be represented here?

Reply:
This mean that it is possible for the shortened result to have more information than the full test output? If we are reporting these to the result, I would expect it to be reported to the full record even if no additional information/context is given.

Reply:
The shortened result will have only the number of successful/failed/warning and a short message because the maximum size of a [Task's Results](https://tekton.dev/vault/pipelines-v0.17.3/tasks/#emitting-results is limited so the information is less than the full json result. The failures items will not shown in shortened tekton task result but they are logged out in tekton task log.

Reply:
I know that we are deferring some decisions (i.e. around `note` content) to HACBS-1598. Can we at least also consider that issue for adding detailed output for these same cases (well, plus WARNING which isn't currently in the issue)? If some information is being populated to the shortened results, I would expect to be able to glean at least that much information from the full results. In this case, it would at least be the presence/count of skips, errors, and warnings. Preferably with more context/information added in -- if possible.

Reply:
I updated the AC of https://issues.redhat.com/browse/HACBS-1598, could you take a look? Thanks so much for your ideas!

Reply:
HACBS-1598 only takes into account the short results. This thread is focused on ensuring that the long results have at least as much detail as the short ones, namely that skips, error, and warnings are represented here as well. Would it be best to create a new issue for this or is it something that can be modified in this ADR?

Reply:
Apologies for the repetition. I might have gotten lost in all the threads I opened up. I think that my hang-up is coming from the fact that we are calling it the full json output. It is not the full output unless we include all possible output including if there are any skips and errors. As it is now, it is more like a detailed successful and failure output.

Reply:
I can see the confusion now. I think that renaming the `Full JSON output` to `Detailed Conftest output` should clear up the confusion here, as that's a more accurate name for it.

Reply:
That would work! I am glad that I found the proper words to convey what was in my head. :D

Reply:
Updated. Thanks!